vnext/Microsoft.ReactNative/ReactNativeAppBuilder.idl

```diff
@@ -0,0 +1,43 @@
+
+import "ReactNativeHost.idl";
+import "ReactInstanceSettings.idl";
+import "IReactPackageProvider.idl";
+
+#include "DocString.h"
+
+namespace Microsoft.ReactNative
+{
+    [experimental]
+    DOC_STRING("This is the builder for creating ReactInstanceSettings.")
+    runtimeclass ReactInstanceSettingsBuilder
+    {
+        ReactInstanceSettingsBuilder();
+
+        // Properties
+        ReactInstanceSettings ReactInstanceSettings {get;};
+
+        // Set Methods
+        ReactInstanceSettingsBuilder SetBundleRootPath(String path);
```

Comment:
minor nit, but do we need to start each method with "Set"?

Reply:
This is just the initial implementation. I agree the namings are bad :). I will sync up with you tomorrow to get this sorted.